Sources/SpeziViews/ViewModifier/ViewState/OperationState.swift

```diff
@@ -0,0 +1,63 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// The ``OperationState`` protocol is used to map the state of an operation, for example a specific action or a more complex task being performed,
+/// to a SpeziView ``ViewState``, describing the current state of a SwiftUI `View`.
+///
+/// Combined with the ``SwiftUI/View/map(state:to:)`` view modifier, the ``OperationState`` provides an easy-to-use
```

Comment:
As stated above, the `OperationState` is not necessarily about just mapping to the `ViewState`. It is a general mental model we want to establish to model operational states within the Spezi framework ecosystem.

I would propose the following structure within the documentation 

```
## Overview
**Short overview description sentence. Explain the general mental model of an operation state: specific action or task (an operation). Well-defined start and end states (e.g. a definite error or result state).**

**Illustrate the concept using a short code example. Like the enum states you already defined.**

### Bridging to ViewState

**Touch how ViewState coexists with a OperationState and how/why a OperationState can be mapped to a ViewState. **

**Code example that demonstrates usage of the `map(state:to:)` modifier and the mapping to `ViewState`.

``

Reply:
Thanks a lot for the insight, adapted my documentation to your suggestions and added a few other points!